Makefile.in

```diff
@@ -31,15 +31,27 @@ dune=@dune@
 # make, since it seems faster.
 #
 # CR-someday mshinwell: We should replace this with a coldstart using dune.
+#
+# The compiler can discard flambda-backend flags from OCAMLPARAM
+# by listing them in "ocaml/stdlib/ocaml_compiler_internal_params".
+# Copy this file to install directory of stage0 to make it available
+# to the build of the compiler itself in stage0 (instead of patching the
+# upstream compiler to look for it in the right place).
+# This is only needed for using OCAMLPARAM with the build of the compiler
+# itself. OCAMLLIB is set to work around the hardcoded library path
+# used by a copy of Compenv in boot/ocamlc.
 .PHONY: stage0
-stage0: _build0/config.status
+stage0: _build0/config.status ocaml_compiler_internal_params
 	rm -f .rsync-output
 	rsync -i -a --filter=':- $$(pwd)/ocaml/.gitignore' \
 	  $$(pwd)/ocaml/ $$(pwd)/_build0 \
 	  | grep -v '/$$' \
 	  | tee .rsync-output
-	if [ -s .rsync-output ] || ! [ -d @stage0_prefix@ ]; then \
+	if [ -s .rsync-output ] || ! [ -d @stage0_prefix@/bin ]; then \
 	  (cd _build0 && \
+	    mkdir -p $(stage0_prefix)/lib/ocaml && \
+	    cp ../ocaml_compiler_internal_params $(stage0_prefix)/lib/ocaml/ && \
+	    export OCAMLLIB=$(stage0_prefix)/lib/ocaml/ && \
```

Comment:
I am out of my depth here, but doesn't it mean that
when triggering a rebuild, the installed files (_e.g._ of
the stdlib) will take precedence over their working-tree
counterpart?

Reply:
Good point, I don't know, but one way to address it would be to remove @stage0_prefix@/lib and rebuild it.

Reply:
I was wondering whether it would make sense
to use a completely different directory, for the
`ocaml_compiler_internal_params` file alone.
Deleting the whole installed `lib/` directory is
likely to work.

Reply:
The location of `ocaml_compiler_internal_param` is hardcoded in `Compenv`. Rather than changing it, we can save and restore `OCAMLPARAM` around `stage0` build. I don't like the hack with `OCAMLLIB`.

Reply:
(When I wrote "a different directory", I meant set
`OCAMLLIB` to something that is not `$PREFIX/lib`.)